infrastructure/dynamo_db.tf

```diff
@@ -96,25 +96,34 @@ module "zip_store_reference_dynamodb_table" {
       type = "S"
     },
     {
-      name = "FileLocation"
+      name = "JobId"
       type = "S"
     },
     {
       name = "NhsNumber"
       type = "S"
-    }
+    },
+    {
+      name = "FileLocation"
+      type = "S"
+    },
   ]
 
   global_secondary_indexes = [
     {
-      name            = "FileLocationsIndex"
-      hash_key        = "FileLocation"
+      name            = "JobIdIndex"
+      hash_key        = "JobId"
       projection_type = "ALL"
     },
     {
       name            = "NhsNumberIndex"
       hash_key        = "NhsNumber"
       projection_type = "ALL"
+    },
+    {
+      name            = "FileLocationsIndex"
+      hash_key        = "FileLocation"
+      projection_type = "ALL"
     }
```

Comment:
I think we could remove the FileLocationsIndex GSI, as we probably don't have a use case for doing a query with the FileLocation string